matrix_client/user.py

```diff
@@ -33,7 +34,11 @@ def get_display_name(self):
             str: Display Name
         """
         if not self.displayname:
-            self.displayname = self.api.get_display_name(self.user_id)
+            try:
+                self.displayname = self.api.get_display_name(self.user_id)
+            except MatrixRequestError as e:
+                if e.code == 404:
```

Comment:
This has a behavior that may not be intended. If the request fails with any other error code, the method still returns `None`. Currently it is equivalent to
```
except MatrixRequestError:
    pass
```
Maybe you wanted to do:
```
except MatrixRequestError as e:
    if e.code != 404:
        raise
```